x-pack/plugins/observability_solution/observability/public/components/custom_threshold/components/alert_details_app_section/groups.tsx

```diff
@@ -5,16 +5,161 @@
+  const {
+    http: {
+      basePath: { prepend },
+    },
+    share: {
+      url: { locators },
+    },
+  } = useKibana().services;
+
+  const apmLocator = useMemo(() => {
+    const baseLocator = locators.get(APM_APP_LOCATOR_ID);
+    if (!baseLocator) return;
+
+    return {
+      ...baseLocator,
+      getRedirectUrl: (params: SerializableRecord) =>
+        baseLocator.getRedirectUrl(getApmLocatorParams(params)),
+      navigate: (params: SerializableRecord) => baseLocator.navigate(getApmLocatorParams(params)),
+    };
+  }, [locators]);
+
+  const infraSourceLinks: Record<string, string> = {
+    [HOST_NAME]: prepend(`${METRICS_DETAILS_PATH}/host`),
+    [CONTAINER_ID]: prepend(`${METRICS_DETAILS_PATH}/container`),
+    [KUBERNETES_POD_ID]: prepend(`${METRICS_DETAILS_PATH}/pod`),
+    [AWS_EC2_INSTANCE]: prepend(`${METRICS_DETAILS_PATH}/awsEC2`),
+    [AWS_S3_BUCKET]: prepend(`${METRICS_DETAILS_PATH}/awsS3`),
+    [AWS_RDS_DATABASES]: prepend(`${METRICS_DETAILS_PATH}/awsRDS`),
+    [AWS_SQS_QUEUE]: prepend(`${METRICS_DETAILS_PATH}/awsSQS`),
+  };
+
+  const hostTimeRange = `assetDetails=(dateRange:(from:'${timeRange.from}',to:'${timeRange.to}'))`;
+  const infraTimeRange = `_a=(time:(from:'${timeRange.from}',to:'${timeRange.to}',interval:>=1m))`;
+
+  const generateInfraSourceLink = ({ field, value }: GroupItem) => {
+    const link =
+      field === HOST_NAME
+        ? `${infraSourceLinks[field]}/${value}?${hostTimeRange}`
+        : `${infraSourceLinks[field]}/${value}?${infraTimeRange}`;
+
+    return (
+      <a href={link} target="_blank">
+        {value}
+      </a>
+    );
+  };
+
+  const generateApmSourceLink = ({ field, value }: GroupItem) => {
+    const serviceName = groups.find((group) => group.field === SERVICE_NAME)?.value;
+
+    let apmLocatorPayload: SerializableRecord = {
+      serviceName,
+      environment: ENVIRONMENT_ALL,
+      rangeFrom: timeRange.from,
+      rangeTo: timeRange.to,
+    };
+
+    if (field === TRANSACTION_NAME) {
+      apmLocatorPayload = {
+        ...apmLocatorPayload,
+        serviceOverviewTab: 'transactions',
+        transactionName: value,
+      };
+    } else if (field === TRANSACTION_TYPE) {
+      apmLocatorPayload = {
+        ...apmLocatorPayload,
+        transactionType: value,
+      };
+    } else if (field === SERVICE_ENVIRONMENT) {
+      apmLocatorPayload = {
+        ...apmLocatorPayload,
+        environment: value,
+      };
+    }
+
+    return (
+      <a href={apmLocator?.getRedirectUrl(apmLocatorPayload)} target="_blank">
+        {value}
+      </a>
+    );
+  };
+
+  const generateNoSourceLink = ({ value }: GroupItem) => {
+    return <strong>{value}</strong>;
+  };
+
+  const generateSourceLink = ({ field, value }: GroupItem) => {
+    return infraSources.includes(field)
+      ? generateInfraSourceLink({ field, value })
+      : apmSources.includes(field)
+      ? generateApmSourceLink({ field, value })
+      : generateNoSourceLink({ field, value });
```

Comment:
Can we avoid nested ternaries? I think it makes it hard to read the code.

Reply:
updated in 1460a3182799a95a2d0a62f9ae07e06375dd4da7